<!DOCTYPE html
   PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
   "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html
   xmlns="http://www.w3.org/1999/xhtml"
   xml:lang="en-US"
   lang="en-US">
<head profile="http://www.w3.org/2005/11/profile">
<link rel="icon"
      type="image/jpg"
      href="infernal_16x16.jpg" />
<meta http-equiv="Content-Type"
      content="text/html;
      charset=UTF-8" />

<title>Infernal: inference of RNA alignments</title>
<link rel="stylesheet" href="infernal.css" type="text/css" media="screen">
</head>



<body>
<table width="100%">
<tr>
 <td align="center">  <img src="infernal-102x149.png"> </td>
 <td align="left" valign="bottom">
    <h1>Infernal: inference of RNA alignments</h1>

<div id="navbar">
        <a href="http://eddylab.org/infernal/">infernal home</a> |
        <a href="http://rfam.xfam.org/">rfam database</a> |
        <a href="http://eddylab.org/">eddy lab</a> |
</div>
  </td>
</tr></table>




<h2> Overview: </h2>

<p>
Infernal ("INFERence of RNA ALignment") is for searching DNA sequence
databases for RNA structure and sequence similarities. It is an
implementation of a special case of profile stochastic context-free
grammars called <em>covariance models</em> (CMs). A CM is like a sequence profile,
but it scores a combination of sequence consensus and RNA secondary structure
consensus, so in many cases, it is more capable of identifying RNA homologs
that conserve their secondary structure more than their primary sequence.
</p>
<p>
The latest release of Infernal is
<a href="http://eddylab.org/infernal/infernal-1.1.4.tar.gz">
1.1.4 [9 Dec 2020].</a>
</p>

<h2> Documentation: </h2>
<ul>
 <li> User's Guide <a href="http://eddylab.org/infernal/Userguide.pdf"> [PDF, 117 pages] </a>.

 <li> <a href="http://eddylab.org/infernal/README.md">
      README </a>
      from the current release.

  <li> <a href="http://eddylab.org/infernal/RELEASE-1.1.4.md">
      Release notes </a>
      for the current release.
</ul>

<h2> Reference: </h2>

<ul>
  <li> The recommended citation for using Infernal 1.1 is
       E. P. Nawrocki and S. R. Eddy,
       <a href="http://eddylab.org/publications.html#Nawrocki13c">
       Infernal 1.1: 100-fold faster RNA homology searches</a>,
       <em>Bioinformatics</em> 29:2933-2935 (2013).
</ul>


<h2> Downloads: </h2>

<ul>
  <li> The current source code: 
       <a href="http://eddylab.org/infernal/infernal-1.1.4.tar.gz">[Infernal 1.1.4 source tarball, 29.5 MB] </a>

  <li> Source with binaries: <br />
       <a href="http://eddylab.org/infernal/infernal-1.1.4-linux-intel-gcc.tar.gz"> 
       [infernal 1.1.4 with Linux/Intel binaries, 46.3 MB] </a> <br />
       <a href="http://eddylab.org/infernal/infernal-1.1.4-macosx-intel.tar.gz">
       [infernal 1.1.4 with MacOSX/Intel binaries, 44.6 MB] </a>
</ul>


<h2> Or, install using a package manager: </h2>

<ul>
  <li> <a href="https://packages.debian.org/search?keywords=infernal">Debian
      package</a>, thanks to Michael Crusoe. If you are using Debian, you can install
    with: <br/> <code>sudo apt-get install infernal infernal-doc</code>

  <li> <a href="https://anaconda.org/bioconda/infernal">Bioconda package</a>,
    thanks to
    <a href="https://github.com/bgruening">Bj&ouml;rn Gr&uuml;ning</a>.
    If you have <code>conda</code>, you can install with: <br/>
    <code>conda install -c bioconda infernal=1.1.2</code>

  <li> <a href="https://github.com/Homebrew/homebrew-science">Homebrew Science package</a>.
       With <code>homebrew</code>, you can install with <br />
       <code>
          brew tap brewsci/bio <br />
          brew install infernal
       </code>
</ul>

Infernal is open source software, freely distributed under the terms
of a standard <a href="http://eddylab.org/infernal/LICENSE"> BSD
(Berkeley Software Distribution) license.</a>.


<h2> Other useful Infernal resources: </h2> 

<ul>
  <li> The Infernal <a href="https://github.com/EddyRivasLab/infernal">GitHub repository</a>.

  <li> <a href="https://quay.io/repository/biocontainers/infernal?tab=tags">
       BioContainer wrapper</a>, thanks to 
       <a href="https://github.com/bgruening">Bj&ouml;rn Gr&uuml;ning</a>.

  <li> <a href="https://toolshed.g2.bx.psu.edu/view/bgruening/infernal/">
       Galaxy wrapper</a>, also thanks to
       <a href="https://github.com/bgruening">Bj&ouml;rn Gr&uuml;ning</a>.
</ul>



<h2> Contact us: </h2>

<ul>
  <li> We welcome bug reports, feature requests, and code contributions. Email us at: 
       <a href="mailto:&#115;&#101;&#097;&#110;&#064;&#101;&#100;&#100;&#121;&#108;&#097;&#098;&#046;&#111;&#114;&#103;">&#115;&#101;&#097;&#110;&#064;&#101;&#100;&#100;&#121;&#108;&#097;&#098;&#046;&#111;&#114;&#103;</a>.</br>
</ul>


<h2> Internal benchmark: </h2>

<ul>
  <li> The Infernal 1.1 Bioinformatics publication contains results
       from our internal RMARK3 benchmark. Files necessary for
       reproducing that benchmark are available <a
       href="http://eddylab.org/infernal/i1.1-benchmark-rmark3.tar.gz">
       here</a>.  </ul>

<h2> Further reference: </h2>

<ul>
 <li> The
      <a href="http://rfam.xfam.org">Rfam database</a> of RNA families
      is based on the Infernal software. 
      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6754622">
      This article</a> demonstrates how to use the Rfam database and
      website and includes a section on using Infernal for genome
      annotation. Rfam 14 is described in 
      <a href="https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkaa1047/5992291">
      (Kalvari, 2020)</a>.

 <li> <a href="https://academic.oup.com/nar/article/46/15/7970/4999243">
      This paper</a> details the discovery of group I introns in archaea
      using Infernal.

 <li> <a href="http://eddylab.org/publications.html#Nawrocki13">
      This book chapter</a> explains how to use Infernal and Rfam to
      annotate RNAs in genomes. 

 <li> Infernal-based identification of functional RNAs in metagenomic
      datasets is discussed in <a href="http://eddylab.org/publications.html#Nawrocki13b">
      (Nawrocki and Eddy, 2013)</a>.

 <li> Version 1.0 of Infernal was announced in a 2009 Bioinformatics 
      <a href="http://eddylab.org/publications.html#Nawrocki09">
      applications note</a>.

 <li> Several improvements in Infernal 1.0, including techniques used
      to improve speed performance, are described in
      <a href="http://eddylab.org/publications.html#Nawrocki09b">
      Eric Nawrocki's 2009 Ph.D. thesis</a>.

 <li> The trCYK local alignment algorithm is described in
      <a href="http://eddylab.org/publications.html#KolbeEddy09">
      (Kolbe and Eddy, 2009)</a>.

 <li> The QDB query-dependent banded dynamic programming algorithm used
      to increase search speed is described in
      <a href="http://eddylab.org/publications.html#NawrockiEddy07">
      (Nawrocki and Eddy, 2007)</a>, along with
      mixture Dirichlet priors and entropy weighting.

 <li> The
      <a href="http://eddylab.org/software.html#rsearch">RSEARCH</a>
      software for searching with single RNA structures (instead of with
      profiles built from multiple alignments) is based on Infernal's
      alignment engines
      <a href="http://eddylab.org/publications.html#KleinEddy03">
      (Klein and Eddy, 2003)</a>. RSEARCH also included MPI parallelization
      and E-value calculations, features that were merged back into
      Infernal in Infernal 1.0.

 <li> The memory-efficient divide-and-conquer CYK alignment algorithm is
      described in
      <a href="http://eddylab.org/publications.html#Eddy02b">
      (Eddy, 2002)</a>. A new implementation, Infernal, replaced COVE.
      
 <li> A chapter of <cite>Biological Sequence Analysis</cite>
      <a href="http://eddylab.org/publications.html#Durbin98">
      (Durbin et al., 1998)</a> describes CMs in more detail.

 <li> The tRNA identification program <a href="http://eddylab.org/software.html#trnascanse">
      tRNAscan-SE</a> is based on a tRNA CM and COVE
      <a href="http://eddylab.org/publications.html#LoweEddy97">
      (Lowe and Eddy, 1997)</a>.

 <li> Covariance models were originally described in
      <a href="http://eddylab.org/publications.html#Eddy94">
      (Eddy and Durbin, 1994)</a>. They were first implemented
      in the
      <a href="http://eddylab.org/software.html#cove">
      COVE</a> software package.
</ul>      




<h2> Archived releases: </h2>

<p> For reproducing published results, the following archived older
versions may be of interest: </p>

<table>
<tr> <th width=50px> Release </th>  <th width=100px> Date </th>  <th width=180px> Download </th>  <th> Notes </th> </tr>

<tr> <th> 1.1.3      </th>
     <td> Nov 2019   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-1.1.3.tar.gz">
          [infernal-1.1.3.tar.gz] </a>
     </td>
     <td> First version where cmscan and cmpress worked for non-calibrated models.
     </td>    
</tr>

<tr> <th> 1.1.2      </th>
     <td> Jul 2016   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-1.1.2.tar.gz">
          [infernal-1.1.2.tar.gz] </a>
     </td>
     <td> First version with cmscan capable of handling long sequences (e.g. chromosomes).
     </td>    
</tr>

<tr> <th> 1.1.1      </th>
     <td> Jul 2014   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-1.1.1.tar.gz">
          [infernal-1.1.1.tar.gz] </a>
     </td>
     <td> Version used for the Rfam 12.0 paper:
      <a href="http://eddylab.org/publications.html#Nawrocki15">
       (Nawrocki, 2015)</a>
     </td>    
</tr>

<tr> <th> 1.1      </th>
     <td> Oct 2013   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-1.1.tar.gz">
          [infernal-1.1.tar.gz] </a>
     </td>
     <td> First stable 1.1 release; the version described in the 2013
     Bioinformatics application note: <a href="http://eddylab.org/publications.html#Nawrocki13c">
     (Nawrocki and Eddy, 2013) </a>
     </td>    
</tr>

<tr> <th> 1.0.2      </th>
     <td> Oct 2009   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-1.0.2.tar.gz">
          [infernal-1.0.2.tar.gz] </a>
     </td>
     <td> Last release before the major 1.1 update. This was the
          "current" version for nearly three years.
     </td>    
</tr>

<tr> <th> 1.0.1      </th>
     <td> Oct 2009   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-1.0.1.tar.gz">
          [infernal-1.0.1.tar.gz] </a>
     </td>
     <td> Version that reproduces the results in Eric Nawrocki's thesis,
          <a href="http://eddylab.org/publications.html#Nawrocki09b">
          (Nawrocki, 2009) </a>
     </td>    
</tr>

<tr> <th> 1.0        </th>
     <td> Jan 2009   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-1.0.tar.gz">
          [infernal-1.0.tar.gz] </a>
     </td>
     <td> Version described in the first "official" release paper,
          <a href="http://eddylab.org/publications.html#Nawrocki09">
          (Nawrocki et al., 2009) </a>
     </td>    
</tr>

<tr> <th> 0.81           </th>
     <td> May 2007   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.81.tar.gz">
          [infernal-0.81.tar.gz] </a>
     </td>
     <td> First version with E-values and MPI parallelization.
     </td>    
</tr>

<tr> <th> 0.72       </th>
     <td> Jan 2007   </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.72.tar.gz">
          [infernal-0.72.tar.gz] </a>
     </td>
     <td> Version used for the published version of the QDB algorithm paper 
          <a href="http://eddylab.org/publications.html#NawrockiEddy07">
          (Nawrocki and Eddy, 2007) </a>
     </td>    
</tr>

<tr> <th> 0.71      </th>
     <td> Nov 2006  </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.71.tar.gz">
          [infernal-0.71.tar.gz] </a>
     </td>
     <td> Version used in the submitted version of the QDB algorithm manuscript
          <a href="http://eddylab.org/publications.html#NawrockiEddy07">
          (Nawrocki and Eddy, 2007) </a>
     </td>    
</tr>

<tr> <th> 0.7       </th>
     <td> Dec 2005  </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.7.tar.gz">
          [infernal-0.7.tar.gz] </a>
     </td>
     <td> Entropy weighting incorporated. This version was used by
          Paul Gardner and Eva Freyhult in their
          <a href="http://www.binf.ku.dk/~pgardner/bralibase/bralibase3/">
          BRaliBaseIII</a> benchmark.
     </td>    
</tr>

<tr> <th> 0.6       </th>
     <td> Nov 2005  </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.6.tar.gz">
          [infernal-0.6.tar.gz] </a>
     </td>
     <td> Mixture Dirichlets incorporated, under pressure from Paul Gardner and
          Eva Freyhult's ongoing benchmarking work.
     </td>    
</tr>

<tr> <th> 0.55      </th>
     <td> Apr 2003  </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.55.tar.gz">
          [infernal-0.55.tar.gz] </a>
     </td>
     <td> First version with documentation. Rfam 2.0 through Rfam 7.0 are based
          on 0.5x code
          <a href="http://eddylab.org/publications.html#Griffiths-Jones05">
          (Griffiths-Jones et al, 2005) </a>
      </td>    
</tr>

<tr> <th> 0.4      </th>
     <td> Aug 2002 </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.4.tar.gz">
          [infernal-0.4.tar.gz] </a>
     </td>
     <td> cmalign first appears, and was used for final work on Rfam 1.0.
     </td>    
</tr>

<tr> <th> 0.3      </th>
     <td> Jul 2002 </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.3.tar.gz">
          [infernal-0.3.tar.gz] </a>
     </td>
     <td> Version used for the first Rfam release
          <a href="http://eddylab.org/publications.html#Griffiths-Jones03">
          (Griffiths-Jones et al, 2003) </a>
     </td>    
</tr>

<tr> <th> 0.1      </th>
     <td> Apr 2002 </td>
     <td> <a href="http://eddylab.org/software/infernal/infernal-0.1.tar.gz">
          [infernal-0.1.tar.gz] </a>
     </td>
     <td> Version used in the divide and conquer algorithm paper
          <a href="http://eddylab.org/publications.html#Eddy02b">
          (Eddy, 2002)</a>
     </td>
</tr>

</table>


<hr>
<table width="100%"><tr>
 <td align="left">
    <address><a href="http://eddylab.org/">Eddy/Rivas Laboratory</a></address>
<!-- hhmts start -->
Last modified: Thu Dec 10 07:53:24 2020
<!-- hhmts end -->
 </td>
</tr></table>

</body> </html>
